public/app/features/trails/AutomaticMetricQueries/units.ts

```diff
@@ -1,5 +1,14 @@
 const DEFAULT_UNIT = 'short';
 
+export function getUnitFromMetric(metric: string) {
+  const metricParts = metric.split('_');
+  if (metricParts.at(-1) === 'bytes' || metricParts.at(-1) === 'seconds') {
+    return getUnit(metricParts.at(-1));
```

Comment:
The `getUnit` function is getting a unit for Grafana panels, which will not be ideal for detail view.

E.g., it will show `s` instead of `seconds`.
![image](https://github.com/grafana/grafana/assets/38694490/40e81562-3805-44a0-8eae-616c69587cdc)

Reply:
Also, would we want to show the Grafana panel unit of "short" when we can't determine a unit?

![image](https://github.com/grafana/grafana/assets/38694490/097d47db-ba3f-4ee9-a6d9-0d29b56433d1)

Reply:
good points, darren - thanks for reviewing! per our pairing session today, i refactored `getUnitFromMetric` to no longer use `getUnit` and will return `null` instead of the `DEFAULT_UNIT` (aka `short`) when we can't determine a unit